proposals/dao/fip_56.ts

```diff
@@ -0,0 +1,224 @@
+    core,
+    tribeUsdCompositeOracle,
+    ethers.constants.AddressZero,
+    ONE_IN_BPS, // $1 Exchange
+    collateralizationOracleWrapper,
+    ONE_IN_BPS, // 100% CR threshold
+    tribeMinter.address,
+    OSM_DURATION
+  );
+
+  await tribeReserveStabilizer.deployTransaction.wait();
+
+  logging && console.log('tribeReserveStabilizer: ', tribeReserveStabilizer.address);
+
+  const daiPSMFactory = await ethers.getContractFactory('PriceBoundPSM');
+
+  const PCVDripControllerFactory = await ethers.getContractFactory('PCVDripController');
+
+  // 3. Deploy DAI PriceBoundPegStabilityModule
+  // PSM will trade DAI between 95 cents and 1.05 cents.
+  // If price is outside of this band, the PSM will not allow trades
+  const daiPSM = await daiPSMFactory.deploy(
+    daiFloorPrice,
+    daiCeilingPrice,
+    {
+      coreAddress: core,
+      oracleAddress: chainlinkDaiUsdOracleWrapper,
+      backupOracle: chainlinkDaiUsdOracleWrapper,
+      decimalsNormalizer: daiDecimalsNormalizer,
+      doInvert: false
+    },
+    daiPSMMintFeeBasisPoints,
+    daiPSMRedeemFeeBasisPoints,
+    daiReservesThreshold,
+    daiFeiMintLimitPerSecond,
+    daiPSMBufferCap,
+    dai,
+    compoundDaiPCVDeposit
+  );
+
+  await daiPSM.deployTransaction.wait();
+
+  logging && console.log('daiPSM: ', daiPSM.address);
+
+  // 4. Deploy DAI PCV Dripper
+  const daiPCVDripController = await PCVDripControllerFactory.deploy(
+    core,
+    compoundDaiPCVDeposit,
+    daiPSM.address,
+    dripFrequency,
+    daiDripAmount,
+    incentiveAmount
+  );
+
+  await daiPCVDripController.deployTransaction.wait();
+
+  logging && console.log('daiPCVDripController: ', daiPCVDripController.address);
+
+  return {
+    tribeMinter,
+    tribeReserveStabilizer,
+    daiPSM,
+    daiPCVDripController
+  } as NamedContracts;
+};
+
+export const setup: SetupUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  logging && console.log('No setup');
+};
+
+export const teardown: TeardownUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  logging && console.log('No teardown');
+};
+
+export const validate: ValidateUpgradeFunc = async (addresses, oldContracts, contracts) => {
+  const role = ethers.utils.id('TRIBE_MINTER_ROLE');
+  const core: Core = contracts.core as Core;
+  const tribe: Tribe = contracts.tribe as Tribe;
+  const tribeMinter: TribeMinter = contracts.tribeMinter as TribeMinter;
+  const tribeReserveStabilizer: TribeReserveStabilizer = contracts.tribeReserveStabilizer as TribeReserveStabilizer;
+
+  expect(await core.hasRole(role, tribeReserveStabilizer.address)).to.be.true;
+  expect(await tribe.minter()).to.be.equal(tribeMinter.address);
+  expect(await tribeMinter.isContractAdmin(tribeReserveStabilizer.address)).to.be.true;
+
+  const { daiPCVDripController, daiPSM, dai, compoundDaiPCVDeposit } = contracts;
+
+  expect(await daiPSM.underlyingToken()).to.be.equal(dai.address);
```

Comment:
We should add a check to ensure that the daiPCVDripController is a PCVController in core.

Reply:
This is checked permissions.json